auth-service/src/main/kotlin/uk/gov/communities/delta/auth/Routing.kt

```diff
@@ -0,0 +1,43 @@
+package uk.gov.communities.delta.auth
+
+import io.ktor.server.application.*
+import io.ktor.server.auth.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import uk.gov.communities.delta.auth.security.DeltaADLdapAuthentication
+import uk.gov.communities.delta.auth.security.DeltaLdapPrincipal
+
+fun Application.configureRouting() {
+
+    routing {
+        get("/") {
+            call.respondText("Hello World!")
+        }
+
+        healthcheckRoute()
+        internalRoutes()
+    }
+}
+
+fun Route.healthcheckRoute() {
+    get("/health") {
+        call.respondText("OK")
+    }
+}
+
+// "Internal" to the VPC, this is enforced by load balancer rules
+fun Route.internalRoutes() {
+    val generateSAMLTokenController = Injection.generateSAMLTokenController()
+
+    authenticate(DeltaADLdapAuthentication.NAME) {
+        route("/auth-internal") {
+            get("/auth-diag") {
+                val principal = call.principal<DeltaLdapPrincipal>(DeltaADLdapAuthentication.NAME)!!
+                call.respond(principal)
+            }
+            post("/generate-saml-token") {
+                generateSAMLTokenController.generateSAMLToken(call)
+            }
```

Comment:
Would it be slightly nicer to declare `fun Route.generateSamlTokenRoutes()` in the Controller and then call `generateSamlTokenRoutes()` here?

Kinda moot when it's just a single route, but I it makes the controller a bit more self contained?

Like in https://ktor.io/docs/structuring-applications.html#group_by_file

Reply:
Definitely can, either as a top level function like that, or as a method on the controller `fun routes(route: Route) {...}`.

It's not a strong preference, but I think I prefer having all the routes in one file like it is now. It would get unwieldy with lots of routes, but for now it's quite neat.